images/clustermesh-apiserver/Dockerfile

```diff
@@ -49,6 +58,11 @@ RUN apt-get update && apt-get install -y binutils-aarch64-linux-gnu binutils-x86
 RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium --mount=target=/root/.cache,type=cache --mount=target=/go/pkg,type=cache \
     ./build-gops.sh
 
+# BUILDPLATFORM is an automatic platform ARG enabled by Docker BuildKit.
+# Represents the plataform where the build is happening, do not mix with
+# TARGETARCH
+FROM --platform=${BUILDPLATFORM} ${ETCD_SERVER_IMAGE} as etcd
```

Comment:
Not an expert here, but I guess we should not specify the platform in this case, because we want to grab the etcd image matching the target arch (as we then copy the binary), and not that of the platform where we are building.

Reply:
Very nice catch. I just tested my locally generated image and it does indeed have a `ARM aarch64` etcd executable in it, even when built with `--platform=linux/amd64`. I'll fix that.